modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/rebuild/IndexProducer.java

```diff
@@ -27,9 +27,10 @@
 public interface IndexProducer {
 
   /**
-   * Re-add all data of this service to the index.
+   * Re-add the data of this service to the index for the given type.
+   * R-add ALL data of this service if null is passed instead.
```

Comment:
But actually probably more like

```suggestion
   * Re-add the data of this service to the index.
   * @param type Limit the data added to the index to the given type. Re-add all data if <code>null</code> is given.
```